tests/driver/screenshot/build.rs

```diff
@@ -0,0 +1,159 @@
+        )?;
+
+        #[cfg(not(feature = "build-time"))]
+        if !generate_macro(&source, &mut output, testcase)? {
+            continue;
+        }
+        #[cfg(feature = "build-time")]
+        generate_source(&source, &mut output, testcase)?;
+
+        for (i, x) in test_driver_lib::extract_test_functions(&source)
+            .filter(|x| x.language_id == "rust")
+            .enumerate()
+        {
+            write!(
+                output,
+                r"
+#[test] fn t_{}() -> Result<(), Box<dyn std::error::Error>> {{
+    use i_slint_backend_testing as slint_testing;
+    {}
+    Ok(())
+}}",
+                i,
+                x.source.replace('\n', "\n    ")
+            )?;
+        }
+    }
+
+    //Make sure to use a consistent style
+    println!("cargo:rustc-env=SLINT_STYLE=fluent");
+
+    println!("cargo:rustc-env=SLINT_EXPERIMENTAL_SYNTAX=true");
+
+    Ok(())
+}
+
+#[cfg(not(feature = "build-time"))]
+fn generate_macro(
+    source: &str,
+    output: &mut std::fs::File,
+    testcase: test_driver_lib::TestCase,
+) -> Result<bool, std::io::Error> {
+    if source.contains("\\{") {
+        // Unfortunately, \{ is not valid in a rust string so it cannot be used in a slint! macro
+        output.write_all(b"#[test] #[ignore] fn ignored_because_string_template() {{}}")?;
+        return Ok(false);
+    }
+    let include_paths = test_driver_lib::extract_include_paths(source);
+    output.write_all(b"slint::slint!{")?;
+    for path in include_paths {
+        let mut abs_path = testcase.absolute_path.clone();
+        abs_path.pop();
+        abs_path.push(path);
+
+        output.write_all(b"#[include_path=r#\"")?;
+        output.write_all(abs_path.to_string_lossy().as_bytes())?;
+        output.write_all(b"\"#]\n")?;
+
+        println!("cargo:rerun-if-changed={}", abs_path.to_string_lossy());
+    }
+    let mut abs_path = testcase.absolute_path;
+    abs_path.pop();
+    output.write_all(b"#[include_path=r#\"")?;
+    output.write_all(abs_path.to_string_lossy().as_bytes())?;
+    output.write_all(b"\"#]\n")?;
+    output.write_all(source.as_bytes())?;
+    output.write_all(b"}\n")?;
+    Ok(true)
+}
+
+#[cfg(feature = "build-time")]
+fn generate_source(
+    source: &str,
+    output: &mut std::fs::File,
+    testcase: test_driver_lib::TestCase,
+) -> Result<(), std::io::Error> {
+    use i_slint_compiler::{diagnostics::BuildDiagnostics, *};
+
+    let include_paths = test_driver_lib::extract_include_paths(source)
+        .map(std::path::PathBuf::from)
+        .collect::<Vec<_>>();
+
+    let mut diag = BuildDiagnostics::default();
+    diag.enable_experimental = true;
+    let syntax_node = parser::parse(source.to_owned(), Some(&testcase.absolute_path), &mut diag);
+    let mut compiler_config = CompilerConfiguration::new(generator::OutputFormat::Rust);
+    compiler_config.include_paths = include_paths;
+    // compiler_config.embed_resources = EmbedResourcesKind::EmbedTextures;
```

Comment:
I think we need that for the fonts

Reply:
@ogoffart you mean in the future?

Reply:
I mean now, for anything that tests things with Text.

Ideally we would actually test both.

But I think the most important is to test the renderer with the embedded image, as this is what we do in MCU

Reply:
I see what you mean. Thank you.